crates/bevy_reflect/bevy_reflect_derive/src/container_attributes.rs

```diff
@@ -0,0 +1,155 @@
+use proc_macro2::{Ident, Span};
+use quote::quote;
+use syn::parse::{Parse, ParseStream};
+use syn::punctuated::Punctuated;
+use syn::token::Comma;
+use syn::{Meta, NestedMeta, Path};
+
+#[derive(Clone)]
+pub enum TraitImpl {
+    NotImplemented,
+    Implemented,
+    Custom(Ident),
```

Comment:
I'm really not sure what this would be used for.

Reply:
Me neither 😅

I'm trying not to mess with _too_ much while cleaning things up here so it's easier to review. But if it would be better to just refactor everything at once, I can look into it!

Reply:
Nah, do what you can easily here, and we can follow-up later.